### PR TITLE
DAOS-11865 control: Rename privileged helpers

### PR DESCRIPTION
**Pull request by mjmac**
The daos_admin helper should be named daos_server_helper
in order to make its purpose clear.

Also renames the daos_firmware helper to daos_firmware_helper.

Required-githooks: true

Signed-off-by: Michael MacDonald <mjmac.macdonald@intel.com>

### CODE REVIEW

File: src/tests/ftest/control/daos_server_helper.py [hunk 1]
```diff
@@ -15,7 +15,7 @@
 
 
 class DaosAdminPrivTest(TestWithServers):
-    """Test class for daos_admin privilege tests.
+    """Test class for daos_server_helper privilege tests.
```
Comment (daltonbohning) [inline src/tests/ftest/control/daos_server_helper.py:18]: The class name should be updated as well

File: src/tests/ftest/control/daos_server_helper.py [hunk 2]
```diff
@@ -24,26 +24,26 @@ class DaosAdminPrivTest(TestWithServers):
...
 
         Test Description:
-            Test daos_admin functionality to perform format privileged
+            Test daos_server_helper functionality to perform format privileged
             operations while daos_server is run as normal user.
 
         :avocado: tags=all,pr,daily_regression
         :avocado: tags=hw,small
-        :avocado: tags=daos_admin,basic
+        :avocado: tags=daos_server_helper,basic
```
Comment (daltonbohning) [inline src/tests/ftest/control/daos_server_helper.py:36]: Optional - While we're here, would you mind updating these tags to the new guidelines? This includes putting `features` tags on line3 and using the test method name as a unique tag
```
        :avocado: tags=all,pr,daily_regression
        :avocado: tags=hw,small
        :avocado: tags=control,basic
        :avocado: tags=daos_server_helper,test_daos_server_helper_format
```

Comment (mjmac) [reply src/tests/ftest/control/daos_server_helper.py:36]: d'oh, missed these comments. will push another fix.

**Review by daltonbohning — COMMENTED** [review]
ftest LGTM. thanks for updating

**Review by jolivier23 — DISMISSED** [review]

**Review by kjacque — APPROVED** [review]

**Review by tanabarr — APPROVED** [review]

**Review by jolivier23 — APPROVED** [review]

**Review by kjacque — APPROVED** [review]

**Pull request merged**